cli/lib/kontena/cli/plugins/install_command.rb

```diff
@@ -1,29 +1,32 @@
-require 'open3'
+require 'kontena/plugin_manager'
 
 module Kontena::Cli::Plugins
   class InstallCommand < Kontena::Command
     include Kontena::Util
     include Kontena::Cli::Common
+    include Kontena::PluginManager::Common
 
     parameter 'NAME', 'Plugin name'
 
     option ['-v', '--version'], 'VERSION', 'Specify version of plugin to install'
     option '--pre', :flag, 'Allow pre-release of a plugin to be installed', default: false
 
-    def execute
-      installed_version = Kontena::PluginManager.instance.installed(name)
+    def installer
+      Kontena::PluginManager::Installer.new(name, pre: pre?, version: version)
+    end
 
-      if installed_version
+    def execute
+      if installed?(name)
         installed = spinner "Upgrading plugin #{name.colorize(:cyan)}" do
-          Kontena::PluginManager.instance.upgrade_plugin(name, pre: pre?)
+          installer.upgrade
```

Comment:
fixed in https://github.com/kontena/kontena/pull/2434/commits/ecaeae3bb1ae0f649b20daa222ff134b3e550907#diff-b7721e55d3ea3e85d3e6115f1cb42e44R42